x-pack/platform/plugins/shared/lens/public/datasources/form_based/form_based_suggestions.ts

```diff
@@ -179,15 +177,15 @@ export function getDatasourceSuggestionsForField(
 // Called when the user navigates from Visualize editor to Lens
 export function getDatasourceSuggestionsForVisualizeCharts(
   state: FormBasedPrivateState,
-  contextLayers: Layer[],
+  contextLayers: NavigateToLensLayer[],
```

Comment:
I haven't fully understood why these Layer is not called `NavigateToLensLayer`. I believe this was used in the "convert to lens" logic but why this is here actually?

Reply:
The `Visualize` context is passed thru the suggestion system via this function and generate a final `LensState` shape that would be eventually shown to the user.
Here I've just updated the type making its name more explicit as `NavigateToLensLayer` instead of the generic `Layer`.